include/circt/Dialect/MSFT/MSFT.td

```diff
@@ -57,14 +57,26 @@ class MSFT_Attr<string name, list<Trait> traits = [],
   let mnemonic = ?;
 }
 
+def RootedInstancePath : MSFT_Attr<"RootedInstancePath"> {
+  let summary = "A path through the instance hierarchy with the root module";
+  let parameters = (ins
+    "FlatSymbolRefAttr":$rootModule, ArrayRefParameter<"StringAttr">:$path);
```

Comment:
Minor, but instead of a FlatSymbolRef and array of String, could this be just one SymbolRefAttr? It seems like that is designed to capture nested symbol refs, which seems like this use case.

On the other hand, if splitting the root and path into separate attributes makes it easier to work with in the parser/printer/c++ code, I am good with the current approach.

Reply:
SymbolRefAttr won't work the way you want here.

Reply:
Besides the wrong semantic meaning, the data types have a slight mis-match. Instance names aren't symbols -- they're strings. While the SymbolRefAttribute uses strings under the hood, we couldn't use the parser (which is what I did previously).

Reply:
Fair enough, I guess I wasn't thinking about this the right way.